pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotTableIdealStateBuilder.java

```diff
@@ -115,14 +117,22 @@ public static void buildLowLevelRealtimeIdealStateFor(PinotLLCRealtimeSegmentMan
     pinotLLCRealtimeSegmentManager.setUpNewTable(realtimeTableConfig, idealState);
   }
 
-  public static int getPartitionCount(StreamConfig streamConfig) {
-    PartitionCountFetcher partitionCountFetcher = new PartitionCountFetcher(streamConfig);
+  /**
+   * Fetches the list of {@link PartitionGroupInfo} for the stream, with the help of the current partitionGroups metadata
+   * This call will only skip partitions which have reached end of life and all messages from that partition have been consumed.
+   * The current partition group metadata is used to determine the offsets that have been consumed for a partition.
+   * The current partition group metadata is also used to know about existing partition groupings which should not be disturbed
+   */
+  public static List<PartitionGroupInfo> getPartitionGroupInfoList(StreamConfig streamConfig,
+      List<PartitionGroupMetadata> currentPartitionGroupMetadataList) {
```

Comment:
`PartitionGroupMetadata` seems to be one per segment. Can you clarify why we need the existing segment metadata in order to find the info about existing partition groups in the stream? 
Also, does `currentPartitionGroupMetadataList` contain the metadata for all segments? Only latest segments? Only consuming segments? Latest online segments?  Some documentation will be useful.

Reply:
Added in the method javadoc